src/rules/importBlacklistRule.ts

```diff
@@ -54,9 +88,140 @@ export class Rule extends Lint.Rules.AbstractRule {
 }
 
 function walk(ctx: Lint.WalkContext<string[]>) {
+    type Options = Array<string | { [moduleName: string]: string[] }>;
+    interface BannedImports {
+        [moduleName: string]: true | Set<string>;
+    }
+
+    // Merge/normalize options.
+    // E.g., ["a", { "b": ["c"], "d": ["e"] }, "f", { "f": ["g"] }]
+    // becomes { "a": true, "b": Set(["c"]), "d": Set(["e"]), "f": true }.
+    const bannedImports = (ctx.options as Options).reduce<BannedImports>(
```

Comment:
Actually, it's probably worth for perf reason to move options normalization to the `Rule` class above as it's not worth to preprocess options for each file separately.